src/sentry/api/serializers/models/event.py

```diff
@@ -346,13 +349,32 @@ class SqlFormatEventSerializer(EventSerializer):
     Applies formatting to SQL queries in the serialized event.
     """
 
+    formatted_sql_cache: Dict[str, str] = {}
+
+    # Various checks to ensure that we don't spend too much time formatting
+    def _should_skip_formatting(self, query: str):
+        if (len(self.formatted_sql_cache) >= MAX_SQL_FORMAT_OPS) | (
+            len(query) > MAX_SQL_FORMAT_LENGTH
+        ):
+            return True
+
+        return False
+
     def _remove_doublequotes(self, message: str):
         return SQL_DOUBLEQUOTES_REGEX.sub(r"\1", message)
 
     def _format_sql_query(self, message: str):
+        formatted = self.formatted_sql_cache.get(message, None)
+        if formatted is not None:
```

Comment:
is there a reason we can't just check `if formatted:` here?

Reply:
Technically `formatted` could be an empty string which would be falsey